tasks.c

```diff
@@ -1050,33 +1051,33 @@ static void prvAddNewTaskToReadyList( TCB_t * pxNewTCB ) PRIVILEGED_FUNCTION;
 
                     uxCoreMap &= ( ( 1 << configNUM_CORES ) - 1 );
 
-                    while( uxCoreMap != 0 )
+                    for( x = configNUM_CORES; x >= 0; x-- )

```

Comment:
Should the loop not be `for( x = 0; x < configNUM_CORES; x++ ) `? Seems like one extra iteration this way.